### DISCUSSION TITLE
The yield! keyword

### DISCUSSION THREAD

**Discussion by HeyAnon** [dotnet/csharplang#4447] [Language Ideas] [answered]
Hi guys, I'm tired of writing a boilerplate every day at work:

    public class A
    {
        public virtual IEnumerable<string> Test()
        {
            yield break;
        }
    }

    public class B : A
    {
        public override IEnumerable<string> Test()
        {
            yield return "aaa";

            foreach (var baseElement in base.Test())
            {
                yield return baseElement;
            }

            yield return "ccc";
        }
    }

I saw a great solution in the F#: https://docs.microsoft.com/ru-ru/dotnet/fsharp/language-reference/sequences#the-yield-keyword 

Code

        public override IEnumerable<string> Test()
        {
            yield return "aaa";

            yield! return base.Test();

            yield return "ccc";
        }

much more compact and readable

**Comment by ufcpp** [comment, marked as answer]
https://github.com/dotnet/csharplang/discussions/378

**Comment by theunrepentantgeek** [comment]
To summarise discussions elsewhere...

Introducing a keyword is believed to imply it's an O(1) operation when the only possible compiler implementation is O(n). This makes the feature extremely likely to conceal performance issues, creating a tiger-trap for future developers. 

Writing the boilerplate might be tedious, but the LDM prefers it to the creation of dangers.

**Reply by NikitaEgorov** [reply]
F# already have this construction  https://docs.microsoft.com/ru-ru/dotnet/fsharp/language-reference/sequences#the-yield-keyword
